lucille-core/src/main/java/com/kmwllc/lucille/stage/ReplacePatterns.java

```diff
@@ -51,15 +57,16 @@ public class ReplacePatterns extends Stage {
   private List<Pattern> patterns;
 
   public ReplacePatterns(Config config) {
-    super(config, new StageSpec().withRequiredProperties("source", "dest", "regex", "replacement")
-        .withOptionalProperties("update_mode", "ignore_case", "multiline", "dotall", "literal"));
+    super(config, new StageSpec().withRequiredProperties("source", "dest", "regex")
+        .withOptionalProperties("replacement", "replacement_field", "update_mode", "ignore_case", "multiline", "dotall", "literal"));
 
     this.patterns = new ArrayList<>();
 
     this.sourceFields = config.getStringList("source");
     this.destFields = config.getStringList("dest");
     this.regexExprs = config.getStringList("regex");
-    this.replacement = config.getString("replacement");
+    this.replacement = ConfigUtils.getOrDefault(config, "replacement", null);
```

Comment:
let's require that the user provide at least one of replacement or replacement_field